payu/models/staged_cable.py

```diff
@@ -0,0 +1,205 @@
+                    cable_stages.append(stage_name)
+
+        # Finish handling of single step stage
+        return cable_stages
+
+    def setup(self):
+        super(StagedCable, self).setup()
+
+        # Directories required by CABLE for outputs
+        os.makedirs(os.path.join(self.work_output_path, 'logs'),
+                    exist_ok=True)
+        os.makedirs(os.path.join(self.work_output_path, 'restart'),
+                    exist_ok=True)
+        os.makedirs(os.path.join(self.work_output_path, 'outputs'),
+                    exist_ok=True)
+
+        self._prepare_stage()
+
+    def _prepare_stage(self):
+        """Apply the stage namelist to the master namelist."""
+
+        if self.configuration_log['current_stage'] != '':
+            # If the current stage is a non-empty string, it means we exited
+            # during the running of the previous stage
+            stage_name = self.configuration_log['current_stage']
+        else:
+            # Pop the stage from the list
+            stage_name = self.configuration_log['queued_stages'].pop(0)
+            self.configuration_log['current_stage'] = stage_name
+
+        with open('configuration_log.yaml', 'w') as conf_log_f:
+            yaml.dump(self.configuration_log, conf_log_f)
+
+        # Ensure the directory exists
+        if not os.path.isdir(stage_name):
+            errmsg = f"""Directory containing namelists for stage {stage_name}
+             does not exist."""
+            raise FileNotFoundError(errmsg)
+
+        # Apply the stage namelists
+        self._apply_stage_namelists(stage_name)
+
+        # Copy the log to the work directory
+        shutil.copy('configuration_log.yaml', self.work_input_path)
+
+    def _apply_stage_namelists(self, stage_name):
+        """Apply the stage namelists to the master namelists."""
+        namelists = os.listdir(stage_name)
+
+        for namelist in namelists:
+            with open(namelist, 'r') as master_nml_f:
+                master_namelist = f90nml.read(master_nml_f)
+
+            with open(os.path.join(stage_name, namelist), 'r') as stage_nml_f:
+                stage_namelist = f90nml.read(stage_nml_f)
+
+            deep_update(master_namelist, stage_namelist)
+
+            # Write the namelist to the work directory
+            master_namelist.write(os.path.join(self.work_input_path, namelist),
+                                  force=True)
+
+    def archive(self):
+        """Store model output to laboratory archive and update the
+configuration log."""
+
+        # Update the configuration log and save it to the working directory
+        completed_stage = self.configuration_log['current_stage']
+        self.configuration_log['current_stage'] = ''
+        self.configuration_log['completed_stages'].append(completed_stage)
+
+        self.save_configuration_log()
+
+        if int(os.environ['PAYU_N_RUNS']) == 1:
+            os.remove('configuration_log.yaml')
```

Comment:
I'm wondering if we can give an error message if a user attempts to set the number of runs with `-n`?

It would probably require inspecting the value of `PAYU_N_RUNS` at the beginning.

Perhaps it is sufficient to print to STDOUT when the number of runs is set, to let them know it is being overridden.